set.mm

```diff
@@ -794139,6 +794317,452 @@ regarding the multiplication ( ~ rngqiprnglin ) via ~ rngqiprnglinlem1
+       23-Mar-2025.) $)
+    pzriprnglem13 $p |- Q e. Ring $=
+      ( vi vx vy wcel co wceq cz c1 crg crng cv cmulr cfv wral wrex c2idl csubg
+      wa cbs pzriprnglem1 pzriprnglem8 pzriprnglem4 cqus oveq2i eqtri qus2idrng
+      cqg eqid mp3an csn cxp ciun zex snex xpex snid xpeq2d sneqd eleq2d rspcev
+      1z mp2an eliun mpbir pzriprnglem11 eleqtrri oveq1 eqeq1d ovanraleqv id wi
+      sneq pzriprnglem12 a1i ralrimiv rspcedvdw ax-mp isringrng mpbir2an ) AUAP
+      AUBPZMUCZNUCZAUDUEZQZWNRZWNWMWOQWNRUJNAUKUEZUFZMWRUGZCUBPECUHUEZPECUIUEPW
+      LCGULCEFGHIUMCEGHUNCEAXAACBUOQCCEUSQZUOQLBXBCUOKUPUQXAUTURVASTVBZVCZWRPZW
+      TXDOSSOUCZVBZVCZVBZVDZWRXDXJPXDXIPZOSUGZTSPXDXDVBZPZXLVMXDSXCVETVFVGVHXKX
+      NOTSXFTRZXIXMXDXOXHXDXOXGXCSXFTWDVIVJVKVLVNOXDSXIVOVPABCDEFOGHIJKLVQVRXEW
+      SXDWNWOQZWNRZWNXDWOQWNRUJZNWRUFMXDWRWQXQNWNWMWNWOWRXDWMXDRWPXPWNWMXDWNWOV
+      SVTWAXEWBXEXRNWRWNWRPXRWCXEABCDEFWNGHIJKLWEWFWGWHWIMNWRAWOWRUTWOUTWJWK $.
+
+    $( Lemma 14 for ~ pzriprng :  The ring unity of the ring ` Q ` .
+       (Contributed by AV, 23-Mar-2025.) $)
+    pzriprnglem14 $p |- ( 1r ` Q ) = ( ZZ X. { 1 } ) $=
+      ( vx vy cz c1 csn cfv wcel wceq cxp cbs cv cmulr co wa wral cur wrex sneq
+      1z xpeq2d sneqd eleq2d id zex snex xpex a1i rspcedvdw ax-mp pzriprnglem11
+      snid ciun eleq2i eliun bitri mpbir pzriprnglem12 pm3.2i crg pzriprnglem13
+      rgen wb eqid isringid mpbi ) OPQZUAZAUBRZSZVSMUCZAUDRZUEWBTWBVSWCUEWBTUFZ
+      MVTUGZUFZAUHRZVSTZWAWEWAVSONUCZQZUAZQZSZNOUIZPOSZWNUKWOWMVSVSQZSZNPOWIPTZ
+      WLWPVSWRWKVSWRWJVROWIPUJULUMUNWOUOWQWOVSOVRUPPUQURVCUSUTVAWAVSNOWLVDZSWNV
+      TWSVSABCDEFNGHIJKLVBVENVSOWLVFVGVHWDMVTABCDEFWBGHIJKLVIVMVJAVKSWFWHVNABCD
+      EFGHIJKLVLMVTAWCWGVSVTVOWCVOWGVOVPVAVQ $.
+  $}
+
+  $( The non-unital ring ` ( ZZring Xs. ZZring ) ` is unital because it has the
+     two-sided ideal ` ( ZZ X. { 0 } ) ` , which is unital, and the quotient of
+     the ring and the ideal is also unital (using ~ ring2idlqusb ).
+     (Contributed by AV, 23-Mar-2025.) $)
+  pzriprng $p |- ( ZZring Xs. ZZring ) e. Ring $=
+    ( vi czring cxps co crg wcel cv cress cqg cqus wa c2idl cfv wrex wtru oveq2
+    cz eleq1d eqid a1i cc0 csn cxp oveq2d anbi12d pzriprnglem8 pzriprnglem7 cur
+    wceq pzriprnglem13 jctir rspcedvdw mptru wb pzriprnglem1 ring2idlqusb ax-mp
+    crng mpbir ) BBCDZEFZUTAGZHDZEFZUTUTVBIDZJDZEFZKZAUTLMZNZVJOVHUTQUAUBUCZHDZ
+    EFZUTUTVKIDZJDZEFZKAVKVIVBVKUIZVDVMVGVPVQVCVLEVBVKUTHPRVQVFVOEVQVEVNUTJVBVK
+    UTIPUDRUEVKVIFOUTVKVLUTSZVKSZVLSZUFTOVMVPVMOUTVKVLVRVSVTUGTVOVNUTVLUHMZVKVL
+    VRVSVTWASVNSVOSUJUKULUMUTURFVAVJUNUTVRUOUTAUPUQUS $.
+
+  $( The ring unity of the ring ` ( ZZring Xs. ZZring ) ` constructed from the
+     ring unity of the two-sided ideal ` ( ZZ X. { 0 } ) ` and the ring unity
+     of the quotient of the ring and the ideal (using ~ ring2idlqus1 ).
+     (Contributed by AV, 24-Mar-2025.) $)
+  pzriprng1 $p |- ( 1r ` ( ZZring Xs. ZZring ) ) = <. 1 , 1 >. $=
```

Comment:
So to be clear, it would have been possible to prove these two theorems directly using `~ xpsringd` and `~ xpsring1d`, but instead they have been proved using `~ ring2idlqusb` and `~ ring2idlqus1` to give an example of their usage, right?

Reply:
Yes, this is the intention of these two theorems, but it may be not clear enough. Maybe I should provide ~ pzriprng and ~ pzriprng1 with the direct proofs, and name the theorems  using ~ring2idlqusb and ~ring2idlqus1 in their proof ~pzriprngALT and  pzriprng1ALT, with the tags  "(Proof modification is discouraged.)" and "(New usage is discouraged.)".

Reply:
Yes - simply putting that explanation in the comments would have been sufficient, but having the two different proofs would be even better of course!